src/NServiceBus.AcceptanceTests/Core/Audit/When_a_message_is_audited.cs

```diff
@@ -41,30 +43,35 @@ public EndpointWithSeparateBodyStorage()
                  });
             }
 
-            public class AuditBodyStorageBehavior : Behavior<IDispatchContext>
+            public class AuditBodyStorageBehavior : Behavior<IAuditContext>
             {
-                public override Task Invoke(IDispatchContext context, Func<Task> next)
+                public override Task Invoke(IAuditContext context, Func<Task> next)
                 {
-                    foreach (var operation in context.Operations)
+                    //body, headers and metadata can be stored separately here
+
+                    context.AuditAction = new ExcludeBodyFromAuditedMessage();
```

Comment:
Would it make sense to have AuditAction just be a delegate instead of a class?  It seems a bit cumbersome to separate the behavior logic from the behavior itself, or, a good bit of boilerplate if we expect this implementation to be the pattern going forward (having an inner class to implement AuditAction).

Reply:
Discussed with @danielmarbach and we think that there are benefits to have a similar mechanism as we have for recoverability and there might also be scenarios where ctor args might be used. Once could also use the type to check "if the action is of type x" that might be useful